### PR TITLE
Don't check for username on alternative config.

### LINKED ISSUE #13: JDBC url without explicit username is not honored.
Hello!
Seems like some uri parameters are lost on their way to flyway
With the following configuration:

```
postgres:
    reactive:
        client:
            uri: postgresql://dbhost:5432/database?user=user&password=pass&currentSchema=current
---
flyway:
    datasources:
        default:
            url: jdbc:${postgres.reactive.client.uri}
```

Flyway migrations are not triggered at all.

if we explicitly specify ``username`` and ``password`` in the config, flyway at least tries to connect but the ``currentSchema`` is lost anyway.
This is also visible in the log:

```
08:54:39.574 [main] INFO  o.f.c.i.license.VersionPrinter - Flyway Community Edition 5.2.1 by Boxfuse
--
  | 08:54:40.658 [main] INFO  o.f.c.i.database.DatabaseFactory - Database: jdbc:postgresql://dbhost:5432/database (PostgreSQL 11.2)
  | 08:54:40.736 [main] WARN  o.f.c.i.database.base.Database - Flyway upgrade recommended: PostgreSQL 11.2 is newer than this version of Flyway and support has not been tested.
  | 08:54:40.767 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [io.micronaut.configuration.dbmigration.flyway.FlywayConfigurationProperties] could not be loaded: Unable to determine current schema as search_path is empty. Set the current schema in currentSchema parameter of the JDBC URL or in Flyway's schemas property.
  | io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.configuration.dbmigration.flyway.FlywayConfigurationProperties] could not be loaded: Unable to determine current schema as search_path is empty. Set the current schema in currentSchema parameter of the JDBC URL or in Flyway's schemas property.
  | at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1164)
  | at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:235)
  | at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2293)
  | at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:199)
  | at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:186)
  | at io.micronaut.runtime.Micronaut.start(Micronaut.java:69)
  | at io.micronaut.runtime.Micronaut.run(Micronaut.java:288)
  | at io.micronaut.runtime.Micronaut.run(Micronaut.java:274)
  | at at.porscheinformatik.scshub.backend.Application.main(Application.java:11)
  | Caused by: org.flywaydb.core.api.FlywayException: Unable to determine current schema as search_path is empty. Set the current schema in currentSchema parameter of the JDBC URL or in Flyway's schemas property.
  | at org.flywaydb.core.internal.database.postgresql.PostgreSQLConnection.doGetCurrentSchema(PostgreSQLConnection.java:64)
  | at org.flywaydb.core.internal.database.base.Connection.getCurrentSchema(Connection.java:92)
  | at org.flywaydb.core.Flyway.prepareSchemas(Flyway.java:1742)
  | at org.flywaydb.core.Flyway.execute(Flyway.java:1663)
  | at org.flywaydb.core.Flyway.migrate(Flyway.java:1341)
  | at io.micronaut.configuration.dbmigration.flyway.AbstractFlywayMigration.runFlyway(AbstractFlywayMigration.java:80)
  | at io.micronaut.configuration.dbmigration.flyway.AbstractFlywayMigration.run(AbstractFlywayMigration.java:70)
  | at io.micronaut.configuration.dbmigration.flyway.AlternativeMigrationRunner.onCreated(AlternativeMigrationRunner.java:62)
  | at io.micronaut.configuration.dbmigration.flyway.AlternativeMigrationRunner.onCreated(AlternativeMigrationRunner.java:40)
  | at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1523)
  | at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2143)
  | at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:1741)
  | at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1162)
  | ... 8 common frames omitted
```
The docs don't mention currentSchema properties.
Note that we:
* don't have datasource properties since we use reactive postgres
* reuse the postgres connection string by property reference

### PR DESCRIPTION
**Pull request by meredrica**
Resolves #13

### CODE REVIEW

**Comment by ilopmar** [comment]
Why don't you want to check for a `username`?

**Comment by ilopmar** [comment]
I'm sorry, I've read your comment on the issue.

**Comment by ilopmar** [comment]
H2 also support defining the username and password on the URL doing `jdbc:h2:~/test;USER=sa;PASSWORD=123`. Could you please add a new test for this case?

**Comment by meredrica** [comment]
here you go. was sick last week so it took a while

File: flyway/src/test/groovy/io/micronaut/configuration/dbmigration/flyway/FlywayConfigurationPropertiesSpec.groovy [hunk 1]
```diff
@@ -134,4 +134,28 @@ class FlywayConfigurationPropertiesSpec extends Specification {
...
     }
+
+    void 'test define flyway database connection via JDBC only and not use Micronaut datasource'() {
+        given:
+        EmbeddedServer server = ApplicationContext.run(EmbeddedServer,
+            ['spec.name'                         : FlywayConfigurationPropertiesSpec.simpleName,
+             'flyway.datasources.books.locations': 'classpath:databasemigrations',
+             'flyway.datasources.books.url'      : 'jdbc:h2:mem:flywayBooksDB3;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE;USER=sa',
+             'flyway.datasources.books.user'     : '',
+             'flyway.datasources.books.password' : '',
```
Comment (ilopmar) [inline flyway/src/test/groovy/io/micronaut/configuration/dbmigration/flyway/FlywayConfigurationPropertiesSpec.groovy]: Please remove `flyway.datasources.books.user` and `flyway.datasources.books.password`

**Comment by ilopmar** [comment]
Can you please sign the CLA? It seems you pushed the new commit with another credentials.

**Comment by meredrica** [comment]
Not exactly sure why the test fails, I don't know my way around groovy. Can someone help out?

**Comment by ilopmar** [comment]
@meredrica Dont' worry about the failing tests. I've make some changes in your PR and I'll rebase and merge once the GORM one is reviewed and merged.

**Comment by ilopmar** [comment]
Rebased master and merged manually. Thanks!

**Comment by meredrica** [comment]
You're welcome :)

**Pull request closed without merging**